llama.cpp

```diff
@@ -663,10 +663,10 @@ struct llama_model_loader {
                          name.c_str(), llama_format_tensor_shape(ne).c_str(), llama_format_tensor_shape(lt.ne).c_str());
         }
 
-        return get_tensor_for(lt, backend);
+        return get_tensor_for(lt, layer, backend);
     }
 
-    struct ggml_tensor * get_tensor_for(llama_load_tensor & lt, ggml_backend backend) {
+    struct ggml_tensor * get_tensor_for(llama_load_tensor & lt, int layer, ggml_backend backend) {
```

Comment:
What if it were like this so the tensors that are not on a layer don't have to specify -1?
```suggestion
    struct ggml_tensor * get_tensor_for(llama_load_tensor & lt, ggml_backend backend, int layer = -1) {
```